pkg/routine/recurringtask_test.go

```diff
@@ -30,14 +30,26 @@ func (h *MockHandler) Do() {
 }
 
 func TestRecurringTask(t *testing.T) {
+	require := require.New(t)
 	h := &MockHandler{Count: 0}
 	ctx := context.Background()
 	ck := clock.NewMock()
 	task := routine.NewRecurringTask(h.Do, 100*time.Millisecond, routine.WithClock(ck))
 	task.Start(ctx)
 	ck.Add(600 * time.Millisecond)
 	task.Stop(ctx)
-	h.mu.RLock()
-	assert.True(t, h.Count >= 5)
-	h.mu.RUnlock()
+	require.Equal(uint(6), h.Count)
+}
+
+func TestRecurringTaskStopBeforeStart(t *testing.T) {
+	require := require.New(t)
+	h := &MockHandler{Count: 0}
+	ctx := context.Background()
+	ck := clock.NewMock()
+	task := routine.NewRecurringTask(h.Do, 100*time.Millisecond, routine.WithClock(ck))
+	task.Stop(ctx)
```

Comment:
`require.Equal(ErrWrongState, task.Stop(ctx))`

Reply:
also i think can add this line to `TestRecurringTask` above, since all other code is the same